cognite/client/_cognite_client.py

```diff
@@ -35,6 +35,14 @@ class CogniteClient:
         timeout (int): Timeout on requests sent to the api. Defaults to 30 seconds.
         token (Union[str, Callable[[], str]]): A jwt or method which takes no arguments and returns a jwt to use for authentication.
             This will override any api-key set.
+        token_url (str): Optional url to use for token generation.
+            This will override the COGNITE_TOKEN_URL environment variable and only be used if both api-key and token are not set.
+        token_client_id (str): Optional client id to use for token generation.
+            This will override the COGNITE_CLIENT_ID environment variable and only be used if both api-key and token are not set.
+        token_client_secret (str): Optional client secret to use for token generation.
+            This will override the COGNITE_CLIENT_SECRET environment variable and only be used if both api-key and token are not set.
+        token_scopes (list): Optional list fo scopes to use for token generation.
```

Comment:
does token_scopes have a sensible default, possible from base_url?

Reply:
I initially went with `base_url` + `/.default`; but was told that it was likely to be different for each of the IDPs.